### ISSUE TITLE
Add a possibility to exclude a unit from regular CS gifts

### ISSUE THREAD

**Issue by adaneslavo** [LoneGazebo/Community-Patch-DLL#10378]
In UCS I have few units that can only be given through lua code (special conditions related to CSs). They are not regular units nor normal UUs. I don't want them to be given by CSs through any means different than production queue in major player cities. Though I tested multiple options and none of them seems to work, so when you are Friend or Ally with Militaristic CS, it has a small chance to give you that unit as a regular gift. What I tested is:
- _NoMinorCivGift_ column in _Units_ table
- _NoMinorGifts_ column in _Units_ table
- _Civilization_UnitClassOverrides_ table with entires for `CIVILIZATION_MINOR`
- Entries in _Policy_UnitClassReplacements_

What I expect is:
- fix the existing columns (_NoMinorCivGift_ or _NoMinorGifts_) if they should exclude these units, but they are not doing that for some reason;
- add new column to the table blocking chosen units from spawning as regular gifts

I would ask you to change this asap, as this is currently the major gamebreaking bug I found in UCS I cannot resolve without stripping the mod from valuable content. Can you help me with that?

**Comment by axatin** [comment]
`IsNoMinorCivGift` should do exactly this. How did you test it? You need to start a new game, existing city-states won't change which units they give.

**Comment by adaneslavo** [comment]
I was playing brand new game when changing one of these. Now I have both set to 1.

**Comment by axatin** [comment]
Set both `MinorCivGift` and `NoMinorCivGift` to 1 in the units table.

**Comment by axatin** [comment]
Yes, there seems to be an issue there. `NoMinorCivGift` doesn't prevent units from being given as regular city-state gifts.

**Comment by adaneslavo** [comment]
> Set both `MinorCivGift` and `NoMinorCivGift` to 1 in the units table.

I don't want them to be gifted at all, so `MinorCivGift` must be set to 0.

**Comment by adaneslavo** [comment]
> Yes, there seems to be an issue there. `NoMinorCivGift` doesn't prevent units from being given as regular city-state gifts.

I thought it was this one that should prevent it in first scenario.

**Comment by axatin** [comment]
> > Set both `MinorCivGift` and `NoMinorCivGift` to 1 in the units table.
> 
> I don't want them to be gifted at all, so `MinorCivGift` must be set to 0.

`MinorCivGift` means the opposite, that the unit can *only* be given by minor civs and not trained regularly. But the advice was incorrect, disregard it.



> I thought it was this one that should prevent it in first scenario.

It prevents them from being selected as the custom unique units given by military city-states ("they know the secrets of ..., therefore they give you this unit"). But it should be added that the unit is also excluded from the regular gifts.

**Comment by adaneslavo** [comment]
Can this be fixed like in the next beta update? This is huge issue for me currently, probably the biggest in the UCS, breaking few abilities.

**Comment by adaneslavo** [comment]
Uh, oh, I cannot wait for the release! Thank you.
